src/core/notifier/event/event.go

```diff
@@ -0,0 +1,121 @@
+package event
+
+import (
+	"time"
+
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/core/notifier"
+	"github.com/goharbor/harbor/src/core/notifier/model"
+)
+
+// Event to publish
+type Event struct {
+	Topic string
+	Data  interface{}
+}
+
+// EventMetadata is the event raw data to be processed
+type EventMetadata interface {
+	Resolve(event *Event) error
+}
+
+// ImageDelMetaData defines images deleting related event data
+type ImageDelMetaData struct {
+	Topic    string
```

Comment:
If it's ImageDelMetadata the Topic can't be others?

Reply:
> If it's ImageDelMetadata the Topic can't be others? And same for other metadata structs.

ImageDelMetadata model is different with others, I think metadata structs is used to collect raw info of event, there is no need to abstract them into unified struct.